src/sema/ast.rs

```diff
@@ -268,6 +268,8 @@ pub struct Function {
     pub symtable: Symtable,
     // What events are emitted by the body of this function
     pub emits_events: Vec<usize>,
+    // For overloaded functions this is the mangled (unique) name.
+    pub abi_name: String,
```

Comment:
My thoughts why I implemented it this way: Having a dedicated field for this is better because an `Option<String>` is confusing. Why is it optional, because it is not needed or because it is not possible to mangle it or another reason? In ABI generation should I use `name` or the mangled name? With `abi_name` it is clear you should always use that as the ABI name during ABI generationl. Aside that I think calling this field `mangled` is way too general and confusing.